src/sdam/events.ts

```diff
@@ -1,21 +1,58 @@
 import type { Document } from '../bson';
+import {
+  SERVER_CLOSED,
+  SERVER_DESCRIPTION_CHANGED,
+  SERVER_HEARTBEAT_FAILED,
+  SERVER_HEARTBEAT_STARTED,
+  SERVER_HEARTBEAT_SUCCEEDED,
+  SERVER_OPENING,
+  TOPOLOGY_CLOSED,
+  TOPOLOGY_DESCRIPTION_CHANGED,
+  TOPOLOGY_OPENING
+} from '../constants';
 import type { ServerDescription } from './server_description';
 import type { TopologyDescription } from './topology_description';
 
 /**
- * Emitted when server description changes, but does NOT include changes to the RTT.
+ * The base export class for all monitoring events published from server discovery and monitoring.
  * @public
  * @category Event
  */
-export class ServerDescriptionChangedEvent {
-  /** A unique identifier for the topology */
+export abstract class ServerDiscoveryAndMonitoringEvent {
```

Comment:
I don't see the introduction of this class in the kickoff, may want to loop the team in on this change since it is public API.

Reply:
I see, I was sort of just following the precedent set by CMAP logging of having an abstract parent class that groups together a set of logging events. I'll ping the team to be sure though.